### DISCUSSION TITLE
_tw and Composer

### DISCUSSION THREAD

**Discussion by tingeka** [gregsullivan/_tw#111] [Q&A]
I'd like to start by saying that I am really enjoying the _tw DX.

Nevertheless, I've got a couple of questions, but it boils down to this: **how do you deal with Composer dependencies?**

Once you zip/bundle, the vendor folder is not included. I like to use some libraries like [Log1x/navi](https://github.com/Log1x/navi) or [htmlburgers/carbon-fields](https://github.com/htmlburger/carbon-fields), but I am not really sure how to tackle this issue, since once you zip/bundled there is no autoload, so no libraries.

 I wouldn't mind having to install the libraries once the theme is deployed, but even then, there is no composer.json neither.

I could modify the zip.js to also copy the vendor, composer.json, and composer-lock.json. Or maybe just the last two.

So the question remains: how do I deal with this scenario?

---

PS: I know I can version-control the whole WP install, and 'globalize' all the dependencies. That would also mean that I can't really work a theme in isolation.

**Comment by gregsullivan** [comment]
Hi Martín—Thanks for posting this!

I definitely recognize that it's not immediately obvious how to approach Composer dependencies meant to be part of the theme itself (as opposed to the included dev dependencies).

Because Composer doesn't support multiple locations for the `vendor` directory in a single `composer.json` file, I think the approach [described here](https://stackoverflow.com/a/68754807) might be a useful one. You could continue using the existing `composer.json` file for dev dependencies, adding a new `composer-theme.json` file with the vendor directory updated to `theme/vendor`.

I've seen mixed advice on simply adding a second `composer.json` file in a nested folder. Because _tw doesn't use the root `vendor/autoload.php` file, conflicts shouldn't be an issue. Multiple `composer.json`-style files in the root folder feels cleaner to me, but it might feel more straightforward over time just to switch directories to `./theme` and run Composer there versus prefixing Composer commands with something like `COMPOSER=composer-theme.json`.

I'd be curious to hear your thoughts! I could see this evolving into a documentation page at some point.
